### DISCUSSION TITLE
Bad default: "Include Git LFS objects in archives" should be enabled by default

### DISCUSSION THREAD

**Discussion by ell1e** [community/community#35250] [Repositories]
`"Include Git LFS objects in archives"` should be enabled by default for all newly created repositories.

Reason: the entire point of "Download as ZIP" is to get a **working** (which means **complete**) copy of the project's main branch without using git. If I need to just use git anyway as a user to get the other half of missing stuff, I don't understand why even bother then with a download function in the UI. And I would wager most repository owners haven't touched that option purely due to not being aware it might be relevant.

**Comment by AF250329** [comment]
I am using `git` and `Github` for years now - never needed `Git LFS` in any project. So why to enable it ?

**Reply by ell1e** [reply]
LFS is always enabled, if you don't use it just move along. This switch affecting the ZIP archive function does nothing for you if you're not having any LFS files in your repository anyway. But if you *do*, it'll fix the archive function, hence why it should be enabled by default.

**Reply by AF250329** [reply]
Mmm..... Could you show me where it is `always enabled` ? What command I should run to see that `LFS` enabled on my repo ?

**Reply by ell1e** [reply]
Install the `git-lfs` client. Then you can immediately do `git lfs install .` in your repo root (which afaik doesn't change/enable something on the server, it just sets up the hooks and LFS server URLs etc in your local repo to point to the storage server so it goes to the right place), and you're set. Pretty sure it's never not enabled on GitHub's side, you can just start pushing in LFS objects any time.

Edit: the switch I mentioned only concerns "Download as ZIP" then actually including what you added via LFS when someone downloads it again and that someone isn't doing a git clone on the command line anyway. I'm guessing GitHub disables it to cut down on traffic, but using LFS but not wanting much traffic is kind of nonsensical. The entire point of LFS is 1. large files (= traffic-heavy, sadly), 2. that others using the repo also actually get them. After all, if they weren't needed in the repository, why would I upload them with LFS in the first place?
